### PR TITLE
fix conflicting bbox setting between pgfsys-dvisvgm and dvisvgm command;

### LINKED ISSUE #1275: Incorrect tikzpicture size when using dvisvgm output driver (PGF/TikZ)
### Brief outline of the bug

I want to produce an SVG file with a node shape, e.g. rectangle. I use the following LaTeX code:

```latex
\def\pgfsysdriver{pgfsys-dvisvgm.def}
\documentclass{article}
\usepackage[usenames]{color}
\pagestyle{empty}
\usepackage{tikz}
\usetikzlibrary{shapes.geometric}
\begin{document}
\pagecolor{white}%
\begin{tikzpicture}
  \node (start) [draw, rectangle, minimum width=24mm, minimum height=16mm] {How are you?};
\end{tikzpicture}
\end{document}
```

and then I use xelatex and then dvisvgm to produce the SVG file:

```
xelatex -no-pdf -interaction nonstopmode -output-directory .\ test1.tex & dvisvgm -n -o .\test1.svg test1.xdv
```

Unfortunately, the produced SVG file has a background object with larger width than the width of the rectangle, resulting in a TikZ picture with larger width. The width is depended on the text width, i.e., if the text is short, the resulting background object width is not longer than the rectangle width, and there is no problem. But when the text width is longer than some critical length, the tikzpicture width is not the same as the drawn object width. Also, I noticed the same problem with the height. In the following example, the problem appears on the top of the tikzpicture:

```latex
\def\pgfsysdriver{pgfsys-dvisvgm.def}
\documentclass{article}
\usepackage[usenames]{color}
\pagestyle{empty}             % do not remove
\usepackage{tikz}
\usetikzlibrary{shapes.geometric}
\begin{document}
\pagecolor{white}%
\begin{tikzpicture}
  \node (point1) [coordinate] {};
  \node at (0,-3cm) (start) [draw, color=black, rectangle, minimum width=24mm, minimum height=16mm] {How are you?};
  \draw (point1) -- (start);
\end{tikzpicture}
\end{document}
```

with the following command to be executed:

```
xelatex -no-pdf -interaction nonstopmode -output-directory .\ test3.tex & dvisvgm -n -o .\test3.svg test1.xdv
```

I am using: Windows 11 MiKTeX-XeTeX 4.10 (MiKTeX 23.5) dvisvgm 3.0.4 pgf 3.1.10

### Minimal working example (MWE)

```latex
\def\pgfsysdriver{pgfsys-dvisvgm.def}
\documentclass{article}
\usepackage[usenames]{color}
\pagestyle{empty}
\usepackage{tikz}
\usetikzlibrary{shapes.geometric}
\begin{document}
\pagecolor{white}%
\begin{tikzpicture}
  \node (start) [draw, rectangle, minimum width=24mm, minimum height=16mm] {How are you?};
\end{tikzpicture}
\end{document}
```

### PR DESCRIPTION
**Pull request by agrahn**
**Motivation for this change**

fixes #1275

**Checklist**

Please [signoff your commits][git-s] to explicitly state your agreement to the [Developer Certificate of Origin][DCO]. If that is not possible you may check the boxes below instead:

- [x] Code changes are licensed under [GPLv2][GPL] + [LPPLv1.3c][LPPL]
- [x] Documentation changes are licensed under [FDLv1.2][FDL]

[git-s]: https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--s
[DCO]: https://developercertificate.org
[GPL]: https://www.gnu.org/licenses/gpl-2.0.html
[LPPL]: https://www.latex-project.org/lppl/lppl-1-3c.txt
[FDL]: https://www.gnu.org/licenses/fdl-1.2.html

### CODE REVIEW

File: doc/generic/pgf/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Added
 
+### Fixed (Alexander Grahn, 2023-09-12) 
+
+- output BBox adjustment in pgfsys-dvisvgm.def, fixes #1275
+
```
Comment (muzimuzhi) [inline doc/generic/pgf/CHANGELOG.md]: Need to adapt to [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) style.

Comment (agrahn) [reply doc/generic/pgf/CHANGELOG.md]: @muzimuzhi  I think I got the Changelog right now, thanks for providing the link.

**Comment by agrahn** [comment]
@hmenke, @muzimuzhi Do you think this PR could be merged? If something is still missing, please let me know.

File: tex/generic/pgf/systemlayer/pgfsys-dvisvgm.def [hunk 2]
```diff
@@ -127,7 +127,7 @@
   \dp#1=0pt%
   \leavevmode%
   \pgf@xa=\pgf@trimleft@final\relax  \ifdim\pgf@xa=0pt \else\kern\pgf@xa\fi%
-  \raise-\pgf@ya\hbox{\ifpgf@sys@svg@inpicture\else\special{dvisvgm:bbox \pgf@sys@tonumber\pgf@picmaxx\space\pgf@sys@tonumber\pgf@picmaxy}\fi\box#1}%
+  \raise-\pgf@ya\hbox{\ifpgf@sys@svg@inpicture\box#1\else\special{dvisvgm:bbox \pgf@sys@tonumber\pgf@picmaxx\space\pgf@sys@tonumber\pgf@picmaxy}\special{dvisvgm:bbox lock}\box#1\special{dvisvgm:bbox unlock}\fi}%
```
Comment (muzimuzhi) [inline tex/generic/pgf/systemlayer/pgfsys-dvisvgm.def]: For shorter code line hence better readability, maybe
```suggestion
  \raise-\pgf@ya\hbox{%
    \ifpgf@sys@svg@inpicture
      \box#1%
    \else
      \special{dvisvgm:bbox \pgf@sys@tonumber\pgf@picmaxx\space\pgf@sys@tonumber\pgf@picmaxy}%
      \special{dvisvgm:bbox lock}%
      \box#1%
      \special{dvisvgm:bbox unlock}%
    \fi}%
```

**Review by muzimuzhi — APPROVED** [review]
Only a soft styling suggestion.

As I said in https://github.com/pgf-tikz/pgf/issues/1275#issuecomment-1715730395, currently most PRs are blocked by #1116, except for simple typo corrections.

File: tex/generic/pgf/systemlayer/pgfsys-dvisvgm.def (hunk 2 above)
Comment (agrahn) [reply tex/generic/pgf/systemlayer/pgfsys-dvisvgm.def]: Thanks for the suggested edit. May I add `%` to protect line endings of lines containing `\ifpgf@...` and `\else`?
... Ok, doesn't seem to be necessary. Will commit the suggestion.

**Review by hmenke — APPROVED** [review]

**Pull request merged**